### DISCUSSION TITLE
Issue 1966: ref struct challenges

### DISCUSSION THREAD

**Discussion by jonlegendre** [xunit/xunit#2144] [General]
Hi Brad and all,
I am considering addressing [1966](https://github.com/xunit/xunit/issues/1966). 
After cursory investigation I am finding that this will likely require a significant amount of code as the _ref struct_ types, such as span<T> and ReadOnlySpan<T> will not be supported by much of the existing code, which is expecting object parameters; [these _ref struct_ types don't support boxing to object](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/struct#ref-struct). For example, the constructors for AssertActualException require objects for the actual and expected parameters, so these _ref struct_ types are not supported. 
Are you expecting significant code for this solution (i.e. creating new methods and classes that can accept these various _ref struct_ type parameters)? Or should I just consider a sort of pseudo-boxing by converting the _ref struct_ entities to string and invoking existing methods? 

Thank you,
Jon

**Comment by bradwilson** [comment]
I have no idea how much code to expect. 😛

**Comment by jonlegendre** [comment]
Brad et al,
I just submitted a pull request for the changes. This ended up being less impactful than I anticipated due to some nifty support for conversions to other types in the memory extensions classes, and some implicit operators and behaviors.
I realized after completing the code that this was not an accepted help wanted change, so no worries if this is rejected, it was a fun bit of code to play with.

**Comment by bradwilson** [comment]
This all got merged, so closing this discussion.
